crates/interpreter/src/exec.rs

```diff
@@ -1064,11 +1057,7 @@ impl<'m> Thread<'m> {
         self.values().drain(values_len - values_cnt .. values_len - arity);
         self.label().values_cnt += arity;
         match kind {
-            LabelKind::Loop(state) => unsafe {
-                self.frame().side_table = state.side_table;
-                self.parser.restore(state.parser_data)
-            },
-            LabelKind::Block | LabelKind::If => self.take_jump(offset),
+            LabelKind::Block | LabelKind::If | LabelKind::Loop => self.take_jump(offset),
```

Comment:
The match can be removed. And actually as a consequence, it seems the `kind` field can be removed from `LabelKind`.